system/lib/pthread/pthread_create.c

```diff
@@ -111,8 +141,23 @@ int __pthread_create(pthread_t *restrict res, const pthread_attr_t *restrict att
   new->tsd = malloc(PTHREAD_KEYS_MAX * sizeof(void*));
   memset(new->tsd, 0, PTHREAD_KEYS_MAX * sizeof(void*));
 
+  //printf("start __pthread_create: %p\n", self);
+  int rtn = __pthread_create_js(new, attrp, entry, arg);
+  if (rtn != 0)
+    return rtn;
+
+  __tl_lock();
+
+  new->next = self->next;
+  new->prev = self;
+  new->next->prev = new;
+  new->prev->next = new;
+
+  __tl_unlock();
```

Comment:
This was introduced in musl commit [`8f11e61`](https://git.musl-libc.org/cgit/musl/commit/?id=8f11e6127fe93093f81a52b15bb1537edc3fc8af) (and [`ba74a42`](https://git.musl-libc.org/cgit/musl/commit/?id=ba74a42cee90c9a4425188a021b6ad8ba80b9468) for `pthread_key_delete`), but I think using a global thread list is not necessary in WASM.

I had a go to avoid this newly added logic, see for example commit https://github.com/emscripten-core/emscripten/pull/13007/commits/bbc1faab026280ffb574bb57b9a54e34bfe402c5, WDYT?

Reply:
Oh, it seems that avoiding the global thread list also fixes the remaining `browser.test_pthread_gcc_atomic*` failures on Chrome.

FWIW, commit 4dfe1d60cfb4aeea41e3ad14bc0b7bfd7f47b085 is easier to test with (since bbc1faab026280ffb574bb57b9a54e34bfe402c5 was based on PR #13007).

Reply:
The linked list of threads is used internally within musl, so think its good to mirror that if we can.   For example, in __pthread_key_delete.

Is there any reason not to maintain it?  (assuming we can fix the remainging failures?)  Is there any reason you think maintaining the list list is less necessary in wasm?

Reply:
Other than this reduces code size and complexity, there was no particular reason for including this commit. It didn't cause any regressions either, but you're right, we probably shouldn't deviate too much from musl.

Hopefully we can fix the remaining test failures so this PR can land (I do wonder why that commit seemed to fix those failures).